fix(windows_certificate): populate thumbprint when filtering by subject

ArmentaRoberto opened the pull request:
- **windows_certificate: populate thumbprint on filtered subject path**
- **windows_certificate: add tests**
- **changelog: add reno note for windows_certificate thumbprint fix**
- **chore(ci): nudge CI**

ArmentaRoberto committed:
windows_certificate: populate thumbprint on filtered subject path

ArmentaRoberto committed:
windows_certificate: add tests

Signed-off-by: ArmentaRoberto <roberto.ahumada@datadoghq.com>

ArmentaRoberto committed:
changelog: add reno note for windows_certificate thumbprint fix

Signed-off-by: ArmentaRoberto <roberto.ahumada@datadoghq.com>

ArmentaRoberto committed:
chore(ci): nudge CI

mrafi97 commented on pkg/collector/corechecks/system/windowscertificate/windows_certificate.go:
```diff
@@ -608,10 +608,18 @@ func findCertificatesInStore(storeHandle windows.Handle, subjectFilters []string
+			certThumbprint, err := getCertThumbprint(certContext)
+			if err != nil {
+				log.Errorf("Error getting certificate thumbprint: %v", err)
+				// keep or 'continue' depending on desired behavior; keeping matches enumerate path semantics only if err is rare
```
Same as last time, let's add a `continue` here, we shouldn't add a certificate if we can't get its thumbprint.

ArmentaRoberto replied:
committed change

ArmentaRoberto committed:
Update windows_certificate.go

Removing unnecessary comment and added logic to not try to add certificate if thumbprint is not retrieveable

ArmentaRoberto committed:
Merge branch 'main' into fix/wincert-thumbprint-filtered
